parsing/parse.mli

```diff
@@ -29,14 +29,28 @@ val expression : Lexing.lexbuf -> Parsetree.expression
 val pattern : Lexing.lexbuf -> Parsetree.pattern
 
 module Longident: sig
-  (** The functions below can be used to parse Longident safely.
-      All variants parse slightly different sets of {!Longident.t}.
+  (** The functions below can be used to parse Longident safely. *)
+
+  val any: Lexing.lexbuf -> Longident.t
+  (**
+     The function [any] is guaranted to parse all subclasses of {!Longident.t}
+     used in OCaml: values, constructors, simple or extended modules, and types
+     or module types.
+
+     However, some input accepted by [any] are not accepted by the
+     compiler. Contrarily to all specialized parsers below,
+     [any] does not enforce that a module path that contains
+     an application ends on a valid identifier for either a type
+     or a module type. For instance, [F(X).(::)] is not accepted by OCaml parser
```

Comment:
Here is an explanation that I would find clearer:

```
In valid OCaml syntax, only value-level identifiers may end with infix operators [Foo.( + )]. On the other hand, in value-level identifiers the module path [Foo] must be simple ([M.N] rather than [F(X)]): functor applications may only appear in type-level identifiers. As a consequence, a path such as [F(X).( + )] is not a valid OCaml identifier, but it is accepted by this function.
```

(Feature request: infix module type operators.)

Reply:
The explanations is better, but I think it warrants updating the first sentence to
```
     However, this function [any] accepts some inputs which are not accepted by the
     compiler, because they combine functor applications and infix operators.
     In valid OCaml syntax, only value-level identifiers may end with infix
     operators [Foo.( + )].
     Moreover, in value-level identifiers the module path [Foo] must be simple
     ([M.N] rather than [F(X)]): functor applications may only appear in type-level identifiers.
     As a consequence, a path such as [F(X).( + )] is not a valid OCaml identifier;
     but it is accepted by this function.
```